server/src/database/archive.rs

```diff
@@ -90,3 +91,88 @@ pub(crate) fn fetch_transactions(
+    response_derives = "Debug"
+)]
+pub struct TransactionQuery;
+
+#[allow(clippy::unwrap_used, clippy::upper_case_acronyms)]
+pub(crate) fn fetch_transactions_graphql(
+    start_time_millis: i64,
+    end_time_millis: i64,
+    mip_key: &str,
+    base_memo: &str, // Pass the base memo as an argument
+) -> Result<Vec<FetchTransactionResult>> {
+    let start_unix_timestamp = start_time_millis / 1000; // Convert milliseconds to seconds
+    let end_unix_timestamp = end_time_millis / 1000;     // Convert milliseconds to seconds
+
+    let start_datetime = OffsetDateTime::from_unix_timestamp(start_unix_timestamp);
+    let end_datetime = OffsetDateTime::from_unix_timestamp(end_unix_timestamp);
+
+    let start_utc_datetime = start_datetime.expect("not a valid time value").to_offset(UtcOffset::UTC);
+    let end_utc_datetime = end_datetime.expect("not a valid time value").to_offset(UtcOffset::UTC);
+
+    let lower_case_memo = base_memo.to_string();
+    let upper_case_memo = base_memo.to_uppercase();
+    let no_lower_case_memo = format!("no {:?}", upper_case_memo);
+    let no_upper_case_memo = format!("NO {:?}", upper_case_memo);
+
+    let variables = transaction_query::Variables {
+        date_time_gte: Some(start_utc_datetime.to_string()),
+        date_time_lte: Some(end_utc_datetime.to_string()),
+        memo1: Some(lower_case_memo),
+        memo2: Some(upper_case_memo),
+        memo3: Some(no_lower_case_memo),
+        memo4: Some(no_upper_case_memo),
+    };
+    let client = Client::new();
+    let response_body: Response<transaction_query::ResponseData> =
+        post_graphql::<TransactionQuery, _>(&client, "https://graphql.minaexplorer.com", variables)
+            .unwrap();
+
+    let txns = response_body.data.unwrap().transactions;
+
+    let txns = txns
+        .into_iter()
+        .flatten()
+        .filter(|tx| tx.to.clone().unwrap() == tx.from.clone().unwrap())
+        .fold(HashMap::new(), |mut map, txn| {
+            // Dedup based on from public key keeping the most recent transaction
+            map.entry(txn.from.clone().unwrap()).or_insert(txn);
+            map
+        })
+        .into_values()
+        .map(|txn: TransactionQueryTransactions| {
+            let timestamp_seconds = match txn.date_time.expect("not a valid time value").parse::<i64>() {
+                Ok(timestamp) => timestamp,
+                Err(_) => 0,
+            };
+            let offset_datetime = OffsetDateTime::from_unix_timestamp(timestamp_seconds);
+
+            FetchTransactionResult {
+                account: txn.to.clone().unwrap(),
+                hash: txn.hash.clone().unwrap(),
+                memo: txn.memo.clone().unwrap(),
+                height: txn.block_height.clone().unwrap(),
+                timestamp: offset_datetime.expect("not a valid time value").unix_timestamp(),
+                status: MinaBlockStatus::Canonical,
+                nonce: 0, // no field `nonce` on type `TransactionQueryTransactions`
```

Comment:
Modify the schema and query graphql file to include it

Reply:
addressed https://github.com/Granola-Team/mina-on-chain-voting/pull/237/commits/dff5a3f7788814873192f9ad3ed81d1c98c22a5a